frontend-cert/js-projects/voting-system/index.js

```diff
@@ -0,0 +1,48 @@
+// Initialize the poll with a Map
+const poll = new Map();
+
+function addOption(option) {
+  if (!poll.has(option)) {
+    poll.set(option, new Set()); // Each option has a Set to track unique voters
+    return `Option "${option}" added to the poll.`;
+  } else {
+    return `Option "${option}" already exists.`;
+  }
+}
+
+function vote(option, voterId) {
+  if (!poll.has(option)) {
+    return `Option "${option}" does not exist.`;
+  }
+  const voters = poll.get(option);
+  if (voters.has(voterId)) {
+    return `Voter ${voterId} has already voted for "${option}".`;
+  } else {
+    voters.add(voterId); // Add voterId to the Set for this option
+    return `Voter ${voterId} voted for "${option}".`;
+  }
+}
+
+// Display poll results
+function displayResults() {
+  console.log("Poll Results:");
+  for (let [option, voters] of poll.entries()) {
+    return `${option}: ${voters.size} votes`;
+  }
+}
```

Comment:
Having this return here will stop the function execution after the first iteration. 

It would be better to have a result string and then append the results to that string and return it outside of the loop if that makes sense

Reply:
Done!